### PR TITLE
[stable/external-dns] Provider independent CA

### LINKED ISSUE #20796: [stable/external-dns] Allow CAs for any provider in the ConfigMap
**Is your feature request related to a problem? Please describe.**
Currently, there is no way to supply CA chains for providers other than designate:

https://github.com/helm/charts/blob/275d0344f38ae4d6626464c02f6869d9c56f4e55/stable/external-dns/templates/configmap.yaml#L7-L9

This means customising the chart for internal use, modifying the provider container to have the certificate chain baked in, or worse IMO: patch the deployment to mount the certificate chain file.

**Describe the solution you'd like**
Allow the ConfigMap <charts/stable/external-dns/templates/configmap.yaml> to contain a generic list of trusted CAs, not specific to a provider.

**Describe alternatives you've considered**
If different provider's TLS verification process works different (eg: can only check one cert, or must contain the full chain), then perhaps extend the current configuration to other providers.

Eg:

```yml
data:
  {{ .Values.designate.customCA.filename }}: |
{{ .Values.designate.customCA.content | indent 4 }}
  {{ .Values.infoblox.customCA.filename }}: |
{{ .Values.infoblox.customCA.content | indent 4 }}
```

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
#### What this PR does / why we need it:
Many organisations have their own internal CA. This chart previously only supported adding a CA chain to the ConfigMap for OpenStack Designate.

This PR makes the CA chain independent of any provider.

#### Which issue this PR fixes
  - fixes #20796

#### Special notes for your reviewer:
Each provider still needs it's own configuration to use the certificate, but now no longer needs CA certs mounted outside of the chart.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for the contribution, before continue, you need to [sign your existing commits](https://github.com/apps/dco), you can find custom instructions in the “Details” link from the below DCO check, also you should increment the chart version in the `Chart.yml` according to [semver](http://semver.org/):

> Given a version number `MAJOR.MINOR.PATCH`, increment the:
> - MAJOR version when you make incompatible API changes,
> - MINOR version when you add functionality in a backwards-compatible manner, and
> - PATCH version when you make backwards-compatible bug fixes.
> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

**Comment by NickLarsenNZ** [comment]
Sorry for the wait - trying to get the branch cleaned up and a whole bunch of OpenStack Designate changes have come in and complicated things that I'm trying to simplify.

**Comment by NickLarsenNZ** [comment]
@juan131 @prydonius would you like to merge this, or close it?

File: stable/external-dns/values.yaml [hunk 1]
```diff
@@ -195,23 +209,9 @@ coredns:
...
-    ##
-    mountPath: "/config/designate"
-    ## Custom CA filename
-    ##
-    filename: "designate-ca.pem"
-
+designate: {}
   ## Use a host path custom CA (optional)
-  ##  This conflicts setting the above customCA to true and chart rendering will fail if you set customCA to true and specify customCAHostPath
+  ##  This conflicts with global customCA settings. Chart rendering will fail if you set customCA to true and specify customCAHostPath
```
Comment (juan131) [inline stable/external-dns/values.yaml:214]: These changes differ from the ones I see in the **values-production.yaml**. Please ensure they're sync

**Review by juan131 — COMMENTED** [review]
Sorry for the late review! I missed this PR

You changes look great! Thanks so much for this great contribution. I just added a small comment and, once you implement it, we're ready to land it.

File: stable/external-dns/values.yaml (hunk 1 above)
Comment (NickLarsenNZ) [reply stable/external-dns/values.yaml:214]: Thanks @juan131.

This should've been done in a different PR that brought in extra Designate settings #20746
I have updated that, and also rebased, fixed author name, and signoffs.

**Comment by NickLarsenNZ** [comment]
@sameersbn @juan131 @prydonius
Shall I just close the PR?

**Comment by juan131** [comment]
Hi @NickLarsenNZ 

Sorry I've been out a few weeks and I missed this. This chart has been deprecated and moved to https://github.com/bitnami/charts/tree/master/bitnami/external-dns. Any contribution must be done in the new repository instead.

Find more information at https://github.com/helm/charts/tree/master/stable/external-dns#this-helm-chart-is-deprecated

**Pull request closed without merging**